### DISCUSSION TITLE
Where can I see a progress report of the project?

### DISCUSSION THREAD

**Discussion by LifeIsStrange** [openworm/OpenWorm#322] [Q&A] [answered]
I would love to know the progress and the current difficulties/blockers

**Comment by mwatts15** [comment, marked as answer]
Hi, @LifeIsStrange. I hesitate to attempt any kind of summary, generally. The Docker container that's built from this Git repository is one artifact that goes through a slice of the tools and frameworks that were built for, made in association with, or otherwise integrated with OpenWorm over the past several years in terms of modeling and simulation (sibernetic, neuroml, c302, WCON format, owmeta/PyOpenWorm, etc.). It doesn't include some other major things like Geppetto and the robotics project...mostly because it would be awkward to include those things in one Docker container, I think.

I won't even pretend to be up on the science side of things. From what I've seen, OpenWorm folks have kept abreast of developments related to C. elegans and shared that work as part of the open science mission of the project. https://openworm.org/science.html has some details, though some parts are probably out-of-date.

Most people who are still contributing on the project mostly have their heads down: they make contribution as and when they can, so things aren't too noisy. You can still see a steady stream of commits if you watch the right repositories. You may also follow the organization as a whole since apparently that's a thing now...

as far as blockers...I think Andrey and Sergey (Sibernetic) were in need of GPUs to push forward with their work, but had trouble finding them at a reasonable price in Fall of last year. For me (owmeta, sometimes Docker, different Python things), I've just been focused on other things.

**Reply by LifeIsStrange** [reply]
Oh Thank you for the answer !
Well your answer is more organisational and technical than about the scientific blockers regarding C.elegans brain understanding but that's still useful thanks.
What I would love to see would be a summarizing blog/paper. For example, [this recent paper](https://www.cell.com/cell/fulltext/S0092-8674(20)31682-2?_returnURL=https%3A%2F%2Flinkinghub.elsevier.com%2Fretrieve%2Fpii%2FS0092867420316822%3Fshowall%3Dtrue) about C.elegans (not from openworm) has made major advances (in metrology, circuitry mapping, high level interpretation).
From the paper it seems that ionotropic/glutamate receptor network remains to be mapped (GABA has been done)

BTW is there video recordings of https://app.genetics-gsa.org/celegans/abstracts21_report/Assignment?lastNameLetter=G ?

**Reply by pgleeson** [reply]
Hi @LifeIsStrange. Yes, @mwatts15 has given a good quick summary of the project, much of the recent work has focussed on getting the Docker image working with the latest elements from the OpenWorm Software stack. Much of it is boring but important tasks getting the latest versions of (Python) packages to work together. Some other recent work, focussed on getting more experimental data into the project with which to constrain the model, includes this for electrophysiological data from worm neurons: https://github.com/openworm/WormsenseLab_ASH, and conversion of a dataset with genetically encoded unique colors for each neuron: https://github.com/openworm/NeuroPAL. 

To be honest, your question highlights the need for a bit of work on our side to more visibly highlight the recent work in the project. One idea we're pursuing is a ~3 monthly newsletter summarising recent developments in the project, which should be a bit more prominent on the website and our communication channels. 

The question you raise about the scientific advances is a good one, and our progress in this area is held back mainly because we still don't have any central academic funding (though we have got a new stream for donations at [OpenCollective](https://opencollective.com/openworm)). 

While software contributions from general open source developers are always welcome, we do need PhD/postdocs with knowledge of the area to get involved. An initiative to address this is the [OpenWorm Studentship](https://openworm.org/studentships.html) program, aimed specifically at these groups. Getting someone on board to dive deep into a dataset like the one you highlight is exactly the type of task this is intended for.

**Reply by LifeIsStrange** [reply]
Thank you, that's a great answer. Actually I put the wrong link in my first comment, the paper I was referring was exactly [neuroPAL](https://www.cell.com/cell/fulltext/S0092-8674(20)31682-2?_returnURL=https%3A%2F%2Flinkinghub.elsevier.com%2Fretrieve%2Fpii%2FS0092867420316822%3Fshowall%3Dtrue) so it's good to hear that you are working with this dataset!

I find it very unfortunate that you lack funding given that understanding how neurons circuits works would be a breakthrough enabling insights that could be backported to the field of artificial intelligence and also has major open problems regarding qualias/consciousness. Brains are so energy and space efficient in terms of compute abilities and accuracy that they in many regards seems to transcend standard computer science and one would even almost believe they bypass the laws of physics... 
Moreover, C.elegans is one of the most simple brains to study, it has an extremely low number of neurons (though how many synapses? can't find a number)) and as such should enable understanding (maybe through bruteforce search ?)
Finally discoveries in the "simple" system that is C.elegans might have implications in medecine/pharmacology too since e.g the neuroPAL paper make the major discovery that 30% of GABA neurons only receives neurotransmitters extra-synaptically ! (One wonder how can that not be chaotic...) This is disruptive, if true then is might be the biggest discovery in neuroscience of the decade!

For all those reasons, each being sufficient, you should receive plenty of funds, however in the irationnal word we live in, this won't happen without a campaign fund/marketing. Though you might consider doing a patreon in addition to opencollective ? 
As I implied, I truly think you should reach google deepmind, openAI, facebook and microsoft AI or even Baidu for them giving you grants, as you are a major long term enabler for them. The field of deep learning is currently in a local minima (transformers) and few believe it but as an expert it seems obvious we are very likely to reach soon a plateau, a new AI winter and only fundamental research like openworm will be able to climb the plateau.

aside:
@pgleeson 
https://www.cell.com/cell/fulltext/S0092-8674(20)31682-2?_returnURL=https%3A%2F%2Flinkinghub.elsevier.com%2Fretrieve%2Fpii%2FS0092867420316822%3Fshowall%3Dtrue#:~:text=We%20found%20low%20Pearson%20correlation%20between%20these%20functional%20and%20structural%20matrices.
> We found low Pearson correlation between these functional and structural matrices.

I was wondering wether using a different coefficient correlation method that Pearson might reveal higher correlation ?
Using this new coefficient correlation method: https://news.ycombinator.com/item?id=29687613
might reveal better correlation.

**Reply by pgleeson** [reply]
Thanks, I agree more funding would be great...
